src/generate.js

```diff
@@ -23,20 +23,20 @@ export const generateCSS = (selector, styleTypes, stringHandlers,
         }
     });
 
-    return (
+    return [
         generateCSSRuleset(selector, declarations, stringHandlers,
-            useImportant) +
-        Object.keys(pseudoStyles).map(pseudoSelector => {
+            useImportant),
+        ...Object.keys(pseudoStyles).map(pseudoSelector => {
             return generateCSSRuleset(selector + pseudoSelector,
                                       pseudoStyles[pseudoSelector],
                                       stringHandlers, useImportant);
-        }).join("") +
-        Object.keys(mediaQueries).map(mediaQuery => {
+        }),
+        ...Object.keys(mediaQueries).map(mediaQuery => {
```

Comment:
How does `styleSheet.insertRule` work with media queries? Here, we sometimes have large blocks of media queries with muliple rules inside them, like
```
@media screen {
    .a { color: red; }
    .b { color: blue; }
}
```
Are those added appropriately?